src/jit/codegencommon.cpp

```diff
@@ -10160,7 +10160,11 @@ bool Compiler::IsMultiRegReturnedType(CORINFO_CLASS_HANDLE hClass)
     structPassingKind howToReturnStruct;
     var_types         returnType = getReturnTypeForStruct(hClass, &howToReturnStruct);
 
+#ifdef _TARGET_ARM64_
+    return (varTypeIsStruct(returnType) && (howToReturnStruct != SPK_PrimitiveType));
+#else
```

Comment:
Probably also need to look for `SPK_EnclosingType`?

Reply:
I don't think so, at least not for now. For the `SPK_EnclosingType` case the `returnType` will be `TYP_INT` or `TYP_LONG`, so we won't pass the first test.